server/commands/write_authzmodel.go

```diff
@@ -29,137 +30,37 @@ func NewWriteAuthorizationModelCommand(
-	allRelations := map[string]struct{}{}
-	typeToRelations := map[string]map[string]struct{}{}
-	for _, td := range tds {
-		typeName := td.GetType()
-		typeToRelations[typeName] = map[string]struct{}{}
-		for relationName := range td.GetRelations() {
-			typeToRelations[typeName][relationName] = struct{}{}
-			allRelations[relationName] = struct{}{}
-		}
-	}
-
-	for _, td := range tds {
-		for relationName, usersetRewrite := range td.GetRelations() {
-			err := isUsersetRewriteValid(allRelations, typeToRelations[td.GetType()], td.GetType(), relationName, usersetRewrite)
-			if err != nil {
-				return err
-			}
-		}
+	model := &typesystem.AuthorizationModel{
+		ID:              id,
+		Version:         schemaVersion,
+		TypeDefinitions: req.GetTypeDefinitions(),
 	}
 
-	return nil
-}
-
-// isUsersetRewriteValid checks if a particular userset rewrite is valid. The first argument is all the relations in
-// the model, the second argument is the subset of relations on the type where the rewrite occurs.
-func isUsersetRewriteValid(allRelations map[string]struct{}, relationsOnType map[string]struct{}, objectType, relation string, usersetRewrite *openfgapb.Userset) error {
-	if usersetRewrite.GetUserset() == nil {
-		return serverErrors.EmptyRewrites(objectType, relation)
+	if err := model.Validate(); err != nil {
+		return nil, err
 	}
 
-	switch t := usersetRewrite.GetUserset().(type) {
-	case *openfgapb.Userset_ComputedUserset:
-		computedUserset := t.ComputedUserset.GetRelation()
-		if computedUserset == relation {
-			return serverErrors.CannotAllowMultipleReferencesToOneRelation
-		}
-		if _, ok := relationsOnType[computedUserset]; !ok {
-			return serverErrors.RelationNotFound(computedUserset, "", nil)
-		}
-	case *openfgapb.Userset_TupleToUserset:
-		tupleset := t.TupleToUserset.GetTupleset().GetRelation()
-		if _, ok := relationsOnType[tupleset]; !ok {
-			return serverErrors.RelationNotFound(tupleset, "", nil)
-		}
-
-		computedUserset := t.TupleToUserset.GetComputedUserset().GetRelation()
-		if _, ok := allRelations[computedUserset]; !ok {
-			return serverErrors.RelationNotFound(computedUserset, "", nil)
-		}
-	case *openfgapb.Userset_Union:
-		for _, child := range t.Union.GetChild() {
-			err := isUsersetRewriteValid(allRelations, relationsOnType, objectType, relation, child)
-			if err != nil {
-				return err
-			}
-		}
-	case *openfgapb.Userset_Intersection:
-		for _, child := range t.Intersection.GetChild() {
-			err := isUsersetRewriteValid(allRelations, relationsOnType, objectType, relation, child)
-			if err != nil {
-				return err
-			}
-		}
-	case *openfgapb.Userset_Difference:
-		err := isUsersetRewriteValid(allRelations, relationsOnType, objectType, relation, t.Difference.Base)
-		if err != nil {
-			return err
-		}
-
-		err = isUsersetRewriteValid(allRelations, relationsOnType, objectType, relation, t.Difference.Subtract)
-		if err != nil {
-			return err
-		}
+	utils.LogDBStats(ctx, w.logger, "WriteAuthzModel", 0, 1)
```

Comment:
I don't think I changed the location of this function. I believe that it is where it is since even if the write fails this log will be sent. I am not sure I agree with that, but 🤷